roles/openshift_cli/tasks/main.yml

```diff
@@ -8,24 +8,35 @@
     docker pull {{ openshift.common.cli_image }}:{{ openshift_image_tag }}
   when: openshift.common.is_containerized | bool
 
-- name: Create /usr/local/bin/openshift cli wrapper
-  template:
-    src: openshift.j2
-    dest: /usr/local/bin/openshift
-    mode: 0755
+- set_fact:
+    container_name: "openshift-cli-{{ 100000 | random }}"
   when: openshift.common.is_containerized | bool
 
-- name: Create client symlinks
+- name: Copy client binaries out of CLI image for use on the host
+  shell: "{{ item }}"
+  with_items:
+  - docker create --name {{ container_name }} {{ openshift.common.cli_image }}:{{ openshift_image_tag }}
+  - docker cp {{ container_name }}:/usr/bin/openshift /usr/local/bin/openshift
+  - docker cp {{ container_name }}:/usr/bin/oc /usr/local/bin/oc
+  - docker rm {{ container_name }}
+  when: openshift.common.is_containerized | bool
+
+- name: Create oadm client symlink
   file:
-    path: "{{ item }}"
+    path: /usr/local/bin/oadm
     state: link
     src: /usr/local/bin/openshift
-  with_items:
-    - /usr/local/bin/oadm
-    - /usr/local/bin/oc
-    - /usr/local/bin/kubectl
   when: openshift.common.is_containerized | bool
 
+- name: Create kubectl client symlink
+  file:
+    path: /usr/local/bin/kubectl
+    state: link
+    src: /usr/local/bin/oc
+  when: openshift.common.is_containerized | bool
+
+- fail:
```

Comment:
This `fail` is intended to be taken out later right? I only noticed it because I was testing this PR and it 'unexpectedly failed' :)


```
TASK [openshift_cli : fail] ****************************************************
task path: /home/tbielawa/rhat/os/openshift-ansible-test/roles/openshift_cli/tasks/main.yml:38
Tuesday 19 July 2016  12:08:03 -0400 (0:00:02.165)       0:04:08.472 **********
fatal: [54.....149]: FAILED! => {"changed": false, "failed": true, "msg": "Failed as requested from task"}
```